plugin/ale.vim

```diff
@@ -125,9 +125,15 @@ let g:ale_close_preview_on_insert = get(g:, 'ale_close_preview_on_insert', 0)
 " This flag can be set to 0 to disable balloon support.
 let g:ale_set_balloons = get(g:, 'ale_set_balloons', has('balloon_eval') && has('gui_running'))
 
+" This flag can be set to 0 to disable popup support.
+let g:ale_set_popups = get(g:, 'ale_set_popups', has('popupwin'))
```

Comment:
Let's remove this setting and add a separate setting for showing `:ALEDetail` messages in popups, and let's turn it off by default. Most of the text I look at for details will fill the screen horizontally.

Update the documentation for `g:ale_cursor_detail` to mention the popup behaviour. It mentions only the `preview-window` at the moment.

It might be easier if you split your pull request into two parts. First add popup support for hover, and then add popup support for detailed messages for problems.